jdaviz/core/helpers.py

```diff
@@ -679,6 +682,82 @@ def data_has_valid_wcs(data, ndim=None):
     return status
 
 
+def return_data_label(app, loaded_object, ext=None, alt_name=None):
+    """
+    Returns a unique data label that can be safely used to load data into data collection.
+
+    Parameters
+    ----------
+    app : `~jdaviz.app.Application`
+        The application-level object used to determine if a data label is unique.
+    loaded_object : str or object
+        The path to a data file or FITS HDUList or image object or Spectrum1D or
+        NDData array or numpy.ndarray.
+    ext : str, optional
+        The extension (or other distinguishing feature) of data used to identify it.
+        For example, "filename[FLUX]" where "FLUX" is the ext value.
+    alt_name : str, optional
+        Alternate names that can be used if none of the options provided are valid.
+
+    Returns
+    -------
+    data_label : str
+        A unique data label that at its root is either given by the user at load time
+        or created by jdaviz using a description of the loaded file's type.
+    """
+    data_label = None
+
+    if loaded_object is None:
+        pass
+    elif isinstance(loaded_object, str):
+        # This is either the full file path or the basename or the user input data label
+        if loaded_object.lower().endswith(('.jpg', '.jpeg', '.png')):
+            file_format = loaded_object.lower().split(".")[-1]
+            data_label = f"{loaded_object}[{file_format}]"
+        else:
+            data_label = f"{loaded_object}"
+    elif isinstance(loaded_object, fits.hdu.hdulist.HDUList):
+        if hasattr(loaded_object, 'file_name'):
+            data_label = f"{loaded_object.file_name}[HDU object]"
+        else:
+            data_label = "Unknown HDU object"
+    elif isinstance(loaded_object, Spectrum1D):
+        data_label = "Spectrum1D"
+    elif isinstance(loaded_object, NDData):
+        data_label = "NDData"
+    elif isinstance(loaded_object, np.ndarray):
+        data_label = "Array"
+
+    if data_label is None and alt_name is not None and isinstance(alt_name, str):
+        data_label = alt_name
+    elif data_label is None:
+        data_label = "Unknown"
+
+    if ext is not None:
+        data_label = data_label + f"[{ext}]"
+
+    data_label = return_unique_name(app, data_label)
+
+    return data_label
+
+
+def return_unique_name(app, data_label):
+    if app is None:
+        raise ValueError("app is None, cannot check for duplicate data labels")
+    if data_label is None:
+        data_label = "Unknown"
+
+    dc = app.data_collection
+    number_of_duplicates = 0
+    for data in dc:
+        if data_label in data.label:
+            number_of_duplicates += 1
+    if number_of_duplicates > 0:
+        data_label = data_label + f" ({number_of_duplicates})"
```

Comment:
I think I expected the first duplicate to append "(2)" rather than "(1)", just from similar patterns in file systems, etc.

Reply:
I just checked my PCs downloads folder and the first duplicate is named something like `filename (1)`.

Reply:
hmmm, ok.  And checking on my linux machine it just does "copy" then "another copy" then "3rd copy" 😂 .  Curious if anyone else has similar instincts, but we can leave it as-is otherwise.

Reply:
I expect (1) on the first duplicate.